api/provider/aws/test_aws/task_list_test.go

```diff
@@ -0,0 +1,154 @@
+package test_aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/golang/mock/gomock"
+	provider "github.com/quintilesims/layer0/api/provider/aws"
+	"github.com/quintilesims/layer0/api/tag"
+	awsc "github.com/quintilesims/layer0/common/aws"
+	"github.com/quintilesims/layer0/common/config/mock_config"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTaskList(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	tags := models.Tags{
+		{
+			EntityID:   "tsk_id1",
+			EntityType: "task",
+			Key:        "name",
+			Value:      "tsk_name1",
+		},
+		{
+			EntityID:   "tsk_id1",
+			EntityType: "task",
+			Key:        "environment_id",
+			Value:      "env_id1",
+		},
+		{
+			EntityID:   "tsk_id1",
+			EntityType: "task",
+			Key:        "arn",
+			Value:      "arn:aws:ecs:region:012345678910:task/arn1",
+		},
+		{
+			EntityID:   "tsk_id2",
+			EntityType: "task",
+			Key:        "name",
+			Value:      "tsk_name2",
+		},
+		{
+			EntityID:   "tsk_id2",
+			EntityType: "task",
+			Key:        "environment_id",
+			Value:      "env_id2",
+		},
+		{
+			EntityID:   "tsk_id2",
+			EntityType: "task",
+			Key:        "arn",
+			Value:      "arn:aws:ecs:region:012345678910:task/arn2",
+		},
+		{
+			EntityID:   "env_id1",
+			EntityType: "environment",
+			Key:        "name",
+			Value:      "env_name1",
+		},
+		{
+			EntityID:   "env_id2",
+			EntityType: "environment",
+			Key:        "name",
+			Value:      "env_name2",
+		},
+	}
```

Comment:
I'm curious what happens in the case where a layer0 environment not named `test` (e.g. `l0-someotherenvironment-env_id1`) is included in the tags db). Would list still return only the clusters with the environment name `test`?

Reply:
So if we add something like this. 
```go
	{
		EntityID:   "env_id3",
		EntityType: "environment",
		Key:        "name",
		Value:      "env_name3",
	},
```
It'll still only return the list of tasks in "test" environment (set environment). At least in testing, however this functionality is achieved by cli via the `l0 task list --all`

Reply:
@jparsons04: I think you're conflating environment ids with the layer0 instance name, e.g. `env_id1` vs. `test`

Reply:
Yep, sorry. I think my original comment can be discarded.